Content.Server/GameTicking/GamePresets/PresetTraitor.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Content.Server.GameTicking.GameRules;
+using Content.Server.Interfaces.Chat;
+using Content.Server.Interfaces.GameTicking;
+using Content.Server.Mobs.Roles;
+using Content.Server.Players;
+using Content.Server.Sandbox;
+using NFluidsynth;
+using Robust.Server.Interfaces.Player;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.IoC;
+using Robust.Shared.Random;
+using Logger = Robust.Shared.Log.Logger;
+
+namespace Content.Server.GameTicking.GamePresets
+{
+    public class PresetTraitor : GamePreset
+    {
+#pragma warning disable 649
+        [Dependency] private readonly ISandboxManager _sandboxManager;
+        [Dependency] private readonly IChatManager _chatManager;
+        [Dependency] private readonly IGameTicker _gameTicker;
+        [Dependency] private readonly IRobustRandom _random;
+#pragma warning restore 649
+
+        public int MinPlayers { get; set; } = 5;
+        public int MinTraitors { get; set; } = 2;
+        public int PlayersPerTraitor { get; set; } = 5;
+
+        public override bool Start(IReadOnlyList<IPlayerSession> readyPlayers)
+        {
+            if (readyPlayers.Count < MinPlayers)
+            {
+                _chatManager.DispatchServerAnnouncement($"Not enough players readied up for the game! There were {readyPlayers.Count} players readied up out of {MinPlayers} needed.");
+                return false;
```

Comment:
Why not throwing an exception instead of returning a boolean that must be checked?

Reply:
Because the GameTicker will default to a "fallback" preset instead of this returns false. In that case, I personally would prefer returning a boolean instead of throwing an exception and having to handle it. I can change it if needed, though.

Reply:
An exception does not prevent the GameTicker to default to something else. One of the advantage of the exception solution is the ability to react differently depending on the error.

But there is not a subjectively better solution, so do as you wish.